# Import libaries

In [4]:
import kagglehub
import pandas as pd

from pathlib import Path

# 0. Load data

In [6]:
path = kagglehub.dataset_download(
    "mlg-ulb/creditcardfraud",
    output_dir="data"
)

DATASET_DIR = Path(path)
DATA_DIR = DATASET_DIR / "creditcard.csv"

print("Path to dataset files:", DATA_DIR)

Path to dataset files: data\creditcard.csv


# 1. Data inspection

In [11]:
df = pd.read_csv(DATA_DIR)

print("Shape:", df.shape)
print("Columns:", list(df.columns))
print(df.head)

Shape: (284807, 31)
Columns: ['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10', 'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20', 'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount', 'Class']
<bound method NDFrame.head of             Time         V1         V2        V3        V4        V5  \
0            0.0  -1.359807  -0.072781  2.536347  1.378155 -0.338321   
1            0.0   1.191857   0.266151  0.166480  0.448154  0.060018   
2            1.0  -1.358354  -1.340163  1.773209  0.379780 -0.503198   
3            1.0  -0.966272  -0.185226  1.792993 -0.863291 -0.010309   
4            2.0  -1.158233   0.877737  1.548718  0.403034 -0.407193   
...          ...        ...        ...       ...       ...       ...   
284802  172786.0 -11.881118  10.071785 -9.834783 -2.066656 -5.364473   
284803  172787.0  -0.732789  -0.055080  2.035030 -0.738589  0.868229   
284804  172788.0   1.919565  -0.301254 -3.249640 -0.557828  2.630515   
284805

### Info
The dataset contains **284,807 observations and 31 features**. The main features are `Time` and `Amount`, while `V1`–`V28` contain anonymized numerical features obtained through **Principal Component Analysis (PCA)**. All features are numerical, making the dataset suitable for machine learning models.